library/src/scala/compiletime/ops/package.scala

```diff
@@ -0,0 +1,30 @@
+package scala.compiletime
+
+import scala.annotation.infix
+
+package object ops {
+  @infix type ==[X <: AnyVal, Y <: AnyVal] <: Boolean
+  @infix type !=[X <: AnyVal, Y <: AnyVal] <: Boolean
+
+  @infix type +[X <: Int, Y <: Int] <: Int
```

Comment:
These ops need prefixes, otherwise it's going to be hard to add `+[S <: String, T <: String]`...

Reply:
Maybe something like this?:

```
package object ops {
  type ==
  type !=
  
  object int {
    ...
  }
   
  object boolean {
    ...
  }
}
```

Reply:
I don't agree. When you write A + B you don't need to think what to import. Same for terms, same for types. In future PR all these operation types should be multipurpose (accept several different supertype combinations). 
If it's better to do `<: AnyVal` or `<: Boolean | Int` I don't know.

Reply:
The issue is that `A + B` means `+[A, B]` and not `A#+[B]`, so we need a way to resolve that `+`...

The goal is to have type equivalants for term operations, and pretending that there is a `+` defined on `AnyVal` doesn't really go into that direction.

But it should be possible to define your own `+`:

```scala
type +[A <: AnyVal, B <: AnyVal] =
  (A, B) match {
    case (Int, Int)       => A obs.int.+ B
    case (String, String) => A obs.string.+ B
    case _ => Error["type error"]
  }
```

Reply:
So you end up with the same thing. I doubt it's a lot better than handling the supported combinations directly in the compiler under the same symbol.

Reply:
The problem with defining `+` both in `ops.int` and `ops.string` is that this seemingly innocent code actually is a compilation error:

```scala
import scala.compiletime.ops.int._
import scala.compiletime.ops.string._

summon[1 + 2]
//       ^ Reference to + is ambiguous
//         it is both imported by import scala.compiletime.ops.int._
//         and imported subsequently by import scala.compiletime.ops.string._
```

An additional problem is that qualified types cannot be used infix:

```scala
import scala.compiletime.ops._
summon[1 int.+ 2]
//          ^ an identifier expected, but '.' found
```

I agree with @OlivierBlanvillain that prefixing overloaded ops helps with the implementation -- it makes it simpler to constant fold if the name analysis phase already has determined which interpretation of `+` we're talking about. However, in the light of the above issues, I also agree with @soronpo that unprefixed ops make for a better user experience. I've managed to find a middle ground in 7b9ccb41aa0807a95168278dc254127feeb71a01, with a match type redirecting to the correct prefixed version, as Olivier suggested.